homeassistant/components/__init__.py

```diff
@@ -156,10 +156,18 @@ def async_handle_core_service(call):
             return
 
         try:
-            yield from conf_util.async_check_ha_config_file(hass)
+            errors = yield from conf_util.async_check_ha_config_file(hass)
         except HomeAssistantError:
             return
 
+        if errors:
```

Comment:
Moved the error reporting to the service, making the check config function more generic.